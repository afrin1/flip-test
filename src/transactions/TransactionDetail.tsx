import React from "react"
import { StyleSheet, Text, View } from "react-native"
import { PrimaryButton } from "../components/Button";
import { colors, content, heading1, heading2, title } from "../theme";
import testID from "../utils/testID";
import { Transaction } from "./types";

type RouteParam = {
  params: Transaction
}

interface TransactionDetailProps {
  route: RouteParam
}

const TransactionDetail = (props: TransactionDetailProps) => {
  const { params } = props.route
  if (params === null || Object.keys(params).length === 0) {
    return <Text>No data</Text>
  }
  const { id,
    senderBank,
    beneficiaryBank,
    beneficiaryName,
    accountNumber,
    amount,
    remark,
    uniqueCode,
    createdAt } = params

  return (
    <View {...testID('transaction-detail-view')} style={styles.root}>
      <View >
        <View style={styles.idRow}>
          <Text style={heading2 as any}>{`ID TRANSAKSI: #${id}`}</Text>
        </View>
        <View style={styles.detailTransactionRow}>
          <Text style={heading2 as any}>{`DETAIL TRANSAKSI`}</Text>
          <PrimaryButton
            title={'Tutup'}
            onPress={() => console.log('pressed')}
          />
        </View>
        <View style={styles.row}>
          <Text style={heading1 as any}>{`${senderBank} -> ${beneficiaryBank}`}</Text>
        </View>
        <View style={styles.detailContainer}>
          <View>
            <Text style={[{ marginBottom: 10 }, title as any]}>{beneficiaryName}</Text>
            <Text style={content as any}>{accountNumber}</Text>
          </View>
          <View>
            <Text style={[{ marginBottom: 10 }, title as any]}>{'NOMINAL'}</Text>
            <Text style={content as any}>{amount}</Text>
          </View>
        </View>
        <View style={styles.detailContainer}>
          <View>
            <Text style={[{ marginBottom: 10 }, title as any]}>{'BERITA TRANSFER'}</Text>
            <Text style={content as any}>{remark}</Text>
          </View>
          <View>
            <Text style={[{ marginBottom: 10 }, title as any]}>{'KODE UNIK'}</Text>
            <Text style={content as any}>{uniqueCode}</Text>
          </View>
        </View>
        <View style={styles.detailContainer}>
          <View>
            <Text style={[{ marginBottom: 10 }, title as any]}>{'WAKTU DIBUAT'}</Text>
            <Text style={content as any}>{createdAt}</Text>
          </View>
        </View>
      </View>
    </View>)
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    backgroundColor: 'white'
  },
  idRow: {
    paddingVertical: 30,
    paddingHorizontal: 20
  },
  detailTransactionRow: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderWidth: 1,
    borderColor: colors.GRAY,
    paddingVertical: 30,
    paddingHorizontal: 20
  },
  row: {
    paddingVertical: 20,
    paddingHorizontal: 20
  },
  detailContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingLeft: 20,
    paddingRight: 60,
    marginVertical: 10
  }
});

export default TransactionDetail