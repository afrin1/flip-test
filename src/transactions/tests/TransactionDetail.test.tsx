import { render } from '@testing-library/react-native'
import React from 'react'
import TransactionDetail from '../TransactionDetail'

describe('Transaction Detail', () => {
  it('should match snapshot', () => {
    const data = {
      id: "FT2366",
      amount: 2629107,
      uniqueCode: 726,
      status: "SUCCESS",
      senderBank: "bni",
      accountNumber: "8379706095",
      beneficiaryName: "Sammy-Jo Mccall",
      beneficiaryBank: "bca",
      remark: "sample remark",
      createdAt: "2021-09-02 20:49:49",
      completedAt: "2021-09-02 20:49:49",
    }
    expect(render(<TransactionDetail route={{ params: {...data} }} />)).toMatchSnapshot()
  })
})