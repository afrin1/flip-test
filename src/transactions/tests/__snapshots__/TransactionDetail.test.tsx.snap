// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction Detail should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
  testID="transaction-detail-view"
>
  <View>
    <View
      style={
        Object {
          "paddingHorizontal": 20,
          "paddingVertical": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        ID TRANSAKSI: #FT2366
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#C5C5C5",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 20,
          "paddingVertical": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        DETAIL TRANSAKSI
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FF6347",
              "fontSize": 15,
              "fontWeight": "bold",
              "margin": 10,
              "textAlignVertical": "center",
            }
          }
        >
          Tutup
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 20,
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        bni -&gt; bca
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 10,
          "paddingLeft": 20,
          "paddingRight": 60,
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "marginBottom": 10,
              },
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Sammy-Jo Mccall
        </Text>
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          8379706095
        </Text>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "marginBottom": 10,
              },
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              },
            ]
          }
        >
          NOMINAL
        </Text>
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          2629107
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 10,
          "paddingLeft": 20,
          "paddingRight": 60,
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "marginBottom": 10,
              },
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              },
            ]
          }
        >
          BERITA TRANSFER
        </Text>
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          sample remark
        </Text>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "marginBottom": 10,
              },
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              },
            ]
          }
        >
          KODE UNIK
        </Text>
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          726
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 10,
          "paddingLeft": 20,
          "paddingRight": 60,
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "marginBottom": 10,
              },
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              },
            ]
          }
        >
          WAKTU DIBUAT
        </Text>
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          2021-09-02 20:49:49
        </Text>
      </View>
    </View>
  </View>
</View>
`;
